### DISCUSSION TITLE
Proxy server listen on port 443 in absence of LB

### DISCUSSION THREAD

**Discussion by harichards** [gravitational/teleport#18925] [Q&A] [answered]
Hello,

I have a Teleport OSS cluster running on AWS with a single proxy node and no load balancer or reverse proxy (e.g., Nginx). My Route53 record points directly to the proxy's public IP.

I would like to be able to access my cluster at `teleport.example.com`. However, the proxy service is not listening on port 443, it's listening on port 3080. Therefore, I must access my cluster at `teleport.example.com:3080` instead.

Is it possible to configure the port that the proxy uses for the web UI? How would I fix this while introducing as little complexity into my cluster as possible?

Thanks!

**Comment by harichards** [comment, marked as answer]
I was able to get the proxy service to listen on 443 by modifying `/etc/teleport.yaml` manually. I would like to be able to set an environment variable to have this done automatically when the EC2 instance is provisioned.

Is this possible?

**Reply by harichards** [reply]
Okay, I think I see the problem. The proxy service is not running as root so it can't bind to 443. I think my options are one of the following:
- port forward using Nginx with SSL pass-through
- port forward using `iptables`
- use an AWS NLB

I've read that the Teleport proxy service is designed for TLS termination. If I only have one proxy instance, do I really need a solution with load balancing capabilities (either Nginx or AWS NLB), or can I get away with `iptables`? Is it safe to expose the proxy service directly to the internet?

**Reply by webvictim** [reply]
It's safe to expose the Teleport proxy directly to the internet if you like.

You can indeed change the config like this but as you say, Teleport will need to be running as `root` to bind to ports < 443:

```yaml
proxy_service:
  web_listen_addr: 0.0.0.0:443
```

You might be able to use capabilities: i.e. grant `CAP_NET_BIND_SERVICE` to the Teleport process to give it that permission while not running as root, but I haven't tested this.

**Reply by harichards** [reply]
Perfect, thanks.
